resteasy-client-microprofile-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java

```diff
@@ -321,6 +310,14 @@ public RestClientBuilder executorService(ExecutorService executor) {
             resteasyClientBuilder.trustStore(null);
             resteasyClientBuilder.keyStore(null, "");
         }
+
+        configureTrustAll(resteasyClientBuilder);
+        // we need to push a disabled SSL context when SSL has been disabled
+        // because otherwise Apache HTTP Client will try to initialize one and will fail
+        if (ImageInfo.inImageRuntimeCode() && !SslContextConfiguration.isSslNativeEnabled()) {
+            resteasyClientBuilder.sslContext(new DisabledSSLContext());
+        }
+
```

Comment:
This is extremely specific to Quarkus so we would need some sort of hook. Maybe a protected hook method that allows to adjust the builder just before we build it?